scripts/navbar_menu.js

```diff
@@ -54,11 +58,22 @@ window.addEventListener('DOMContentLoaded', function() {
             }
         });
 
+        document.getElementById('site_search').addEventListener('click', showSiteSearch);
```

Comment:
I think the new behavior triggers the default handler that tries to navigate to the `<a>` tag's `href` attribute. The inline `onclick` handler would prevent this default handling through the use of `return false`.

It seems to me that we could remove the need for setting `href='#'` by changing this to:
```
document.getElementById('site_search').addEventListener('click', function (ev) {
  ev.preventDefault(); // Prevents clicking on the anchor.
  showSiteSearch();
});
```

This would be equivalent to the current code and seems slightly better but could be easily convinced it's better to use the current approach.

Reply:
> I think the new behavior triggers the default handler that tries to navigate to the `<a>` tag's `href` attribute. The inline `onclick` handler would prevent this default handling through the use of `return false`.
> 
> It seems to me that we could remove the need for setting `href='#'` by changing this to:
> 
> ```
> document.getElementById('site_search').addEventListener('click', function (ev) {
>   ev.preventDefault(); // Prevents clicking on the anchor.
>   showSiteSearch();
> });
> ```
> 
> This would be equivalent to the current code and seems slightly better but could be easily convinced it's better to use the current approach.

I tried this now with `href = ''`. It does work and is closer to the old behaviour in that if the page is scrolled down slightly it does not jump up to the top as it does with `href='#'`.

Reply:
> Note, if you remove the `href="#"` then you cannot tab to the link (unless you get funky with a `tabindex` attribute). To get open in new tab feature working, we could give the fragment a name like `href="#search"` or something, and have some `window.hashchange` event logic. Just some ideas.

I'm inclined to leave it like it is. We could open a new PR to address the issue of what happens if you try to open the search in a new tab.

Reply:
> I'm inclined to leave it like it is. We could open a new PR to address the issue of what happens if you try to open the search in a new tab, or to prevent this option from being available (e.g by using a button instead of an anchor).

Just verifying (though looking at the code I'm pretty sure) that this is the site search, not the project search?

Reply:
You are right, the site search is the only one which is not really a link.